File: DSA/linked_list/linked_list_find.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None

a = Node('A')
b = Node('B')
c = Node('C')
d = Node('D')

a.next = b
b.next = c
c.next = d

def find(node, target):
    while node:
        if node.val == target:
            return True
        node = node.next

    return False

def find_rec(node, target):
    if node == None:
        return False
    if node.val == target:
        return True
    return find_rec(node.next, target)

if __name__ == '__main__':
    print('Searching for target')
    print(find(a, 'C'))
    print('Searching for target recursively')
    print(find(a, 'D'))

File: dsa/linked_list/linkedListFind.js
class Node {
	constructor(val) {
		this.val = val
		this.next = null
	}
}

const a = new Node('A')
const b = new Node('B')
const c = new Node('C')
const d = new Node('D')

a.next = b
b.next = c
c.next = d

const find = (node, target) => {
	while (node != null) {
		if (node.val === target) return true
		node = node.next
	}
	return false
}

const findRec = (node, target) => {
	if (node === null) return false
	if (node.val === target) return true
	return findRec(node.next, target)
}

console.log('Searching for target')
console.log(find(a, 'C'))
console.log('Searching for target recursively')
console.log(findRec(a, 'D'))
	
